### PR TITLE
feat: add card selection props

### PR DESCRIPTION
**Pull request by pyphilia**

### CODE REVIEW

File: src/Card/Card.tsx [hunk 1]
```diff
@@ -14,20 +16,28 @@ const DEFAULT_CARD_HEIGHT = 130;
...
+    : isSelected
+      ? `2px solid ${PRIMARY_COLOR}`
+      : 'none',
+}));
 
 type CardProps = {
   name: string | JSX.Element;
   alt: string;
   id?: string;
+  className?: string;
```
Comment (spaenleh) [inline src/Card/Card.tsx:40]: Why the use of `className` ?

File: src/Card/Card.tsx [hunk 2]
```diff
@@ -14,20 +16,28 @@ const DEFAULT_CARD_HEIGHT = 130;
 
 const StyledCard = styled(MuiCard, {
   shouldForwardProp: (prop) => prop !== 'elevation' && prop !== 'fullWidth',
```
Comment (spaenleh) [inline src/Card/Card.tsx:18]: I think you need to add some props here to remove warnings that they are not supported on the DOM. `isSelected` and `isOver` for example

**Review by spaenleh — CHANGES_REQUESTED** [review]

File: src/Card/Card.tsx (hunk 1 above)
Comment (pyphilia) [reply src/Card/Card.tsx:40]: I needed a generic way to select all cards, not based on the id https://github.com/graasp/graasp-builder/blob/f7cc3a22b17a8bbf4b587ebd93b1eb34243cc221/src/components/main/list/SelectionContext.tsx#L95

**Review by spaenleh — APPROVED** [review]

File: src/Card/Card.tsx [hunk 3]
```diff
@@ -13,21 +15,26 @@ import CardThumbnail, { CardThumbnailProps } from './CardThumbnail';
...
-  ({ theme, elevation, fullWidth, isOver }) => ({
-    borderRadius: theme.spacing(1),
-    boxShadow: elevation ? theme.shadows[2] : '0px 2px 2px #eeeeee',
-    width: fullWidth ? '100%' : 'max-content',
-    maxWidth: '100%',
-    border: isOver ? '2px solid black' : 'none',
-  }),
-);
+  shouldForwardProp: (prop: string) =>
+    !['elevation', 'fullWidth', 'isSelected', 'isOver'].includes(prop),
```
Comment (ReidyT) [inline src/Card/Card.tsx:19]: Should we extract the array in a const ?

```suggestion
    !PROPS_TO_FORWARD.includes(prop),
```

**Review by ReidyT — APPROVED** [review]
Nice PR, LGTM!

File: src/Card/Card.tsx [hunk 4]
```diff
@@ -13,21 +15,26 @@ import CardThumbnail, { CardThumbnailProps } from './CardThumbnail';
...
+  isOver: boolean;
+  fullWidth?: boolean;
+  elevation?: boolean;
+  isSelected?: boolean;
+}>(({ theme, elevation, fullWidth, isOver, isSelected }) => ({
+  borderRadius: theme.spacing(1),
+  boxShadow: elevation ? theme.shadows[2] : '0px 2px 2px #eeeeee',
+  width: fullWidth ? '100%' : 'max-content',
+  maxWidth: '100%',
+  border: isOver || isSelected ? `2px solid ${PRIMARY_COLOR}` : 'none',
```
Comment (spaenleh) [inline src/Card/Card.tsx:30]: Using `outline` instead of `border` will be better as the element will not change size, removing small layout shifts.

**Review by spaenleh — CHANGES_REQUESTED** [review]

**Review by spaenleh — APPROVED** [review]
LGTM ! 🎈

**Pull request merged**